easybuild/easyblocks/x/xalt.py

```diff
@@ -0,0 +1,92 @@
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for XALT, implemented as an easyblock
+@author: Scott McMillan (NVIDIA)
+"""
+import os
+import re
+
+from easybuild.easyblocks.generic.configuremake import ConfigureMake
+from easybuild.framework.easyconfig import CUSTOM, MANDATORY
+from easybuild.tools.build_log import EasyBuildError
+
+
+class EB_XALT(ConfigureMake):
+    """Support for building and installing XALT."""
+
+    @staticmethod
+    def extra_options():
+        extra_vars = {
+            'config_py': [None, "XALT site filter file", MANDATORY],
+            'syshost': ['hardcode:cluster', "System name", CUSTOM],
+            'transmission': ['syslog', "", CUSTOM],
+        }
+        return ConfigureMake.extra_options(extra_vars)
+
+    def configure_step(self):
+        """Custom configuration step for XALT."""
+
+        # Remove the subdir from the install prefix to resolve the
+        # following configure error.  XALT automatically creates a
+        # versioned directory hierarchy.
+        #
+        #    Do not install XALT with 2.7.27 as part of the
+        #    prefix.  It will lead to many many problems with future
+        #    installs of XALT
+        #
+        #    Executables built with the current version of XALT
+        #    will not work with future installs of XALT!!!
+        #
+        #    If you feel that you know better than the developer
+        #    of XALT then you can configure XALT with the configure
+        #    --with-IrefuseToInstallXALTCorrectly=yes and set the prefix
+        #    to include the version
+        self.installdir = os.path.normpath(re.sub(self.install_subdir, '', self.installdir))
+
+        # XALT site filter config file is mandatory
+        if not self.cfg['config_py']:
+            raise EasyBuildError('XALT site filter config must be specified. '
+                                 'Use "--try-amend=config_py=<file>"')
+        self.cfg.update('configopts', '--with-config=%s' % self.cfg['config_py'])
+
+        if self.cfg['syshost']:
+            self.cfg.update('configopts', '--with-syshostConfig=%s' % self.cfg['syshost'])
+
+        if self.cfg['transmission']:
+            self.cfg.update('configopts', '--with-transmission=%s' % self.cfg['transmission'])
+
```

Comment:
I think it should include GPU tracking support, with something like
```
if get_software_root('cuda'):
        txt += self.module_generator.set_environment('XALT_GPU_TRACKING', 'yes')
    else:
        txt += self.module_generator.set_environment('XALT_GPU_TRACKING', 'no')
```

Reply:
That would require that `XALT` gets installed with `CUDA` listed as a dependency, which I don't think is wise...

@rtmclay told me what XALT actually needs is `nvidia.h` to be available the system, so if we can figure out a way to check for that, we can auto-enable GPU tracking based on that (unless `gpu_tracking = False` is used to override the auto-detection).

Reply:
The latest commit tries to find CUDA using `get_software_root('CUDA')` and falls back to `/usr/local/cuda` is the former is not defined.  It's not perfect (the CUDA module must be loaded prior to running `eb`).  Please let me know if there are better approaches.

Reply:
Checking whether CUDA is loaded via `get_software_root('CUDA')` is fine, but I don't think the fallback to `/usr/local/cuda` makes sense...

Users can easily opt-in to GPU tracking if they really want it when not providing CUDA via EasyBuild?

Reply:
[We do install XALT](https://github.com/eth-cscs/production/blob/da8e1f42dcfe66511778750101d8770b08513962/easybuild/easyconfigs/x/xalt/xalt-2.8.eb#L73) with CUDA as a build dependency and it works. CUDA does not need to be present at run time.